bitnami/wordpress/templates/_helpers.tpl

```diff
@@ -259,16 +259,40 @@ Return the MariaDB User
 {{- end -}}
 
 {{/*
-Return the MariaDB User
+Return the MariaDB Secret Name
 */}}
 {{- define "wordpress.databaseSecretName" -}}
 {{- if .Values.mariadb.enabled }}
     {{- printf "%s" (include "mariadb.fullname" .) -}}
+{{- else if .Values.externalDatabase.existingSecret -}}
+    {{- printf "%s" .Values.externalDatabase.existingSecret -}}
 {{- else -}}
     {{- printf "%s-%s" .Release.Name "externaldb" -}}
 {{- end -}}
 {{- end -}}
 
+{{/*
+Return the WordPress Secret Name
+*/}}
+{{- define "wordpress.secretName" -}}
+{{- if .Values.existingSecret }}
+    {{- printf "%s" .Values.existingSecret -}}
+{{- else -}}
+    {{- printf "%s" (include "wordpress.fullname" .) -}}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Return the SMTP Secret Name
+*/}}
+{{- define "wordpress.smtpSecretName" -}}
+{{- if .Values.smtpExistingSecret }}
+    {{- printf "%s" .Values.smtpExistingSecret -}}
+{{- else -}}
+    {{- printf "%s" (include "wordpress.fullname" .) -}}
+{{- end -}}
+{{- end -}}
```

Comment:
Why do we need a second secret I mean, why not using the same "existing secret" both for SMTP and WP password?

Reply:
I worked Fab with this. We took this route simply to provide more flexibility. If somebody decided to use an existing secret for the smtp password, but either wanted to use the automatically generated WordPress password or just don't care if it's specified at install.

You can use 3 separate secrets for each of these, or you could use 2 or 1 even. I used this new version already as a test and actually used a single secret for the WordPress secret, SMTP, and MariaDB instead of separating the db secret from the others.

Reply:
I don't have a strong opinion about this. I guess that using one secret per credentials might be more flexible, although harder to maintain.
Feel free to use 3 secrets.